components/org.wso2.carbon.identity.api.server.input.validation/org.wso2.carbon.identity.api.server.input.validation.v1/src/main/java/org/wso2/carbon/identity/api/server/input/validation/v1/core/ValidationRulesManagementApiService.java

```diff
@@ -152,51 +151,22 @@ private List<ValidatorModel> buildValidatorResponse(List<ValidatorConfiguration>
     }
 
     /**
-     * Get default configurations.
+     * Get configurations from user store.
      *
-     * @return  default configurations.
+     * @return configurations.
      */
-    private List<ValidationConfigModel> getDefaultConfiguration() {
-
-        // Build default configs.
-        List<ValidationConfigModel> configurations = new ArrayList<>();
-
-        ValidationConfigModel configuration = new ValidationConfigModel();
-        configuration.setField(PASSWORD);
-        List<RuleModel> rules = new ArrayList<>();
-
-        rules.add(getRuleModel("LengthValidator", MIN_LENGTH, "8"));
-        rules.add(getRuleModel("NumeralValidator", MIN_LENGTH, "1"));
-        rules.add(getRuleModel("UpperCaseValidator", MIN_LENGTH, "1"));
-        rules.add(getRuleModel("LowerCaseValidator", MIN_LENGTH, "1"));
-        rules.add(getRuleModel("SpecialCharacterValidator", MIN_LENGTH, "1"));
-        configuration.setRules(rules);
+    private List<ValidationConfigModel> getConfigurationsFromUserStore(String tenantDomain) {
 
-        configurations.add(configuration);
-        return configurations;
-    }
-
-    /**
-     * Get rule model.
-     *
-     * @param validatorName Validator name.
-     * @param property      Property name.
-     * @param value         Property value.
-     *
-     * @return  rule model.
-     */
-    private RuleModel getRuleModel(String validatorName, String property, String value) {
-
-        RuleModel rule = new RuleModel();
-        rule.setValidator(validatorName);
-        List<MappingModel> properties = new ArrayList<>();
-        MappingModel mapping = new MappingModel();
-        mapping.setKey(property);
-        mapping.setValue(value);
-        properties.add(mapping);
-        rule.setProperties(properties);
-
-        return rule;
+        try {
+            List<ValidationConfiguration> configurations = InputValidationServiceHolder.getInputValidationMgtService()
+                    .getConfigurationFromUserStore(tenantDomain);
+            if (configurations != null) {
+                return buildResponse(configurations);
+            }
+        } catch (InputValidationMgtException e) {
+            throw new RuntimeException(e);
```

Comment:
Any specific reason tp throw runtime exception here? Can we avoid throwing runtime exceptions and handle the error sceanro?

Reply:
Addressed with f60a469c0aeadde21d52759ea105e6ec28dff7a1